node/setup.go

```diff
@@ -70,7 +70,7 @@ func DefaultGenesisDocProviderFunc(config *cfg.Config) GenesisDocProvider {
 		if err != nil {
 			return ChecksummedGenesisDoc{}, err
 		}
-		return ChecksummedGenesisDoc{GenesisDoc: genDoc, Sha256Checksum: incomingChecksum}, nil
+		return ChecksummedGenesisDoc{GenesisDoc: genDoc, Sha256Checksum: sizedByteArray(incomingChecksum)}, nil
```

Comment:
Could you explain how the fixed-size array would be better if it introduced all these conversions?

Reply:
yea realised that as well. As a library user its a minor nit to be able to pass what is expected instead, sha256sum, not a unbounded checksum. if the naming is changed to checksum then this becomes mute.

Reply:
The fun part is that the `tmhash.Sum(jsonBlob)` above does exactly the opposite: produces a `[32]int` array and converts it to a slice...